Extension/src/LanguageServer/referencesModel.ts

```diff
@@ -0,0 +1,93 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+import * as vscode from 'vscode';
+import { ReferenceType, ReferenceInfo } from './references';
+
+export class Model {
+    readonly items: FileItem[] = [];
+
+    constructor(resultsInput: ReferenceInfo[]) {
+        let results: ReferenceInfo[] = resultsInput.filter(r => r.type !== ReferenceType.Confirmed);
+        for (let r of results) {
+            // Add file if it doesn't exist
+            let fileItem: FileItem;
+            let index: number = this.items.findIndex(function(item): boolean {
+                return item.name === r.file;
+             });
+            if (index < 0) {
+                const uri: vscode.Uri = vscode.Uri.file(r.file);
+                fileItem = new FileItem(uri, r.file);
+                this.items.push(fileItem);
+            } else {
+                fileItem = this.items[index];
+            }
+
+            // Add reference to file
+            let isFileReference: boolean = (r.position.line === 0 && r.position.character === 0);

```

Comment:
Can you handle "file references"? Those are cases where the line/column are not known yet, in which case, it seems like it should add a link to just the start of the file.

Reply:
We still need to handle file references.